data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/CracImpl.java

```diff
@@ -197,9 +197,13 @@ public Set<State> getStates(Instant instant) {
     @Override
     public State getState(Contingency contingency, Instant instant) {
         Objects.requireNonNull(contingency, "Contingency must not be null when getting a state.");
+        State state = states.get(contingency.getId() + " - " + instant.toString());
+        if (!Objects.isNull(state)) {
+            return state;
+        }
         return states.values().stream()
-            .filter(state -> state.getInstant() == instant)
-            .filter(state -> state.getContingency().isPresent() && state.getContingency().get().getId().equals(contingency.getId()))
+            .filter(s -> s.getInstant() == instant)
+            .filter(s -> s.getContingency().isPresent() && s.getContingency().get().getId().equals(contingency.getId()))
             .findAny()
             .orElse(null);
```

Comment:
why do you still need both?

Reply:
I left it as a precaution in case for some reason we don't create the state with that kind of ID.